src/core/common/message.cpp

```diff
@@ -51,8 +51,10 @@
 #error "OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE conflicts with OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT."
 #endif
 
-#if OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE && !OPENTHREAD_CONFIG_DTLS_ENABLE
-#error "OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE is strongly discouraged when OPENTHREAD_CONFIG_DTLS_ENABLE is off."
+#if OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE && !OPENTHREAD_CONFIG_DTLS_ENABLE && \
+    !OPENTHREAD_CONFIG_NON_THREAD_COMMISSIONING
```

Comment:
Is `OPENTHREAD_CONFIG_NON_THREAD_COMMISSIONING` an existing config? I could not find it. If it is new we should add it and document its purpose and behavior (in one of the `src/config/*.h` files).

But I would suggest not adding a new config.  I think the idea is to silence this `#error`. It may be better to simply remove it (removing this check and error fully from the code to allow `USE_HEAP_ENABLE` to be used with or without `DTLS_ENABLE`).

Reply:
`OPENTHREAD_CONFIG_NON_THREAD_COMMISSIONING ` was added by me. Initially I wanted to just remove the error but I thought it's not a good idea. I will do this then.

Thank you